source/lib/src/rocm/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# required cmake version
+cmake_minimum_required(VERSION 3.5)
+# project name
+project(deepmd_op_rocm)
+set(CMAKE_LINK_WHAT_YOU_USE TRUE)
+
+# set c++ version c++11
+SET(CMAKE_CXX_STANDARD 11)
+SET(CMAKE_HIP_STANDARD 11)
+
+message(STATUS "HIP major version is " ${HIP_VERSION_MAJOR})
+
+set (HIP_HIPCC_FLAGS -hc; -fno-gpu-rdc; --amdgpu-target=gfx906; -fPIC; -O3; --std=c++11)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -DCUB_IGNORE_DEPRECATED_CPP_DIALECT -DCUB_IGNORE_DEPRECATED_CPP_DIALECT")
```

Comment:
Recommend using `add_definitions` rather than hacking `CMAKE_CXX_FLAGS` with `-DCUB_IGNORE_DEPRECATED_CPP_DIALECT -DCUB_IGNORE_DEPRECATED_CPP_DIALECT`

Reply:
Ok, I willly modify later, by the way why do you recommend using add_definitions? Is it because add_definition can affect C++ and C compilers?

Reply:
Does `add_definiations` only affects the building tasks in source/lib/src/rocm? Could you please check if it has affects on building tasks outside rocm?